Packs/FeedUnit42v2/Integrations/FeedUnit42v2/FeedUnit42v2.py

```diff
@@ -0,0 +1,657 @@
+    return 'ok'
+
+
+def fetch_indicators(client: Client, feed_tags: list = [], tlp_color: Optional[str] = None,
+                     create_relationships=False) -> List[Dict]:
+    """Retrieves indicators from the feed
+
+    Args:
+        client: Client object with request
+        feed_tags: feed tags.
+        tlp_color: Traffic Light Protocol color.
+        create_relationships: Create indicators relationships
+    Returns:
+        List. Processed indicators from feed.
+    """
+    item_types_to_fetch_from_api = ['report', 'indicator', 'malware', 'campaign', 'attack-pattern', 'relationship',
+                                    'course-of-action', 'intrusion-set']
+    client.get_stix_objects(items_types=item_types_to_fetch_from_api)
+
+    for type_, objects in client.objects_data.items():
+        demisto.info(f'Fetched {len(objects)} Unit42 {type_} objects.')
+
+    id_to_object = {
+        obj.get('id'): obj for obj in
+        client.objects_data['report'] + client.objects_data['indicator'] + client.objects_data['malware']
+        + client.objects_data['campaign'] + client.objects_data['attack-pattern']
+        + client.objects_data['course-of-action'] + client.objects_data['intrusion-set']
+    }
+
+    ioc_indicators = parse_indicators(client.objects_data['indicator'], feed_tags, tlp_color)
+    reports = parse_reports_and_report_relationships(client.objects_data['report'], feed_tags, tlp_color, id_to_object)
+    campaigns = parse_campaigns(client.objects_data['campaign'], feed_tags, tlp_color)
+    attack_patterns = create_attack_pattern_indicator(client.objects_data['attack-pattern'], feed_tags, tlp_color)
+    intrusion_sets = create_intrusion_sets(client.objects_data['intrusion-set'], feed_tags, tlp_color)
+    course_of_actions = create_course_of_action_indicators(client.objects_data['course-of-action'],
+                                                           feed_tags, tlp_color)
+
+    dummy_indicator = {}
+    if create_relationships:
+        list_relationships = create_list_relationships(client.objects_data['relationship'], id_to_object)
+
+        dummy_indicator = {
+            "value": "$$DummyIndicator$$",
+            "relationships": list_relationships
+        }
+
+    if dummy_indicator:
+        ioc_indicators.append(dummy_indicator)
+
+    if ioc_indicators:
+        demisto.debug(f'{len(ioc_indicators)} XSOAR Indicators were created.')
+    if reports:
+        demisto.debug(f'{len(reports)} XSOAR Reports Indicators were created.')
+    if campaigns:
+        demisto.debug(f'{len(campaigns)} XSOAR campaigns Indicators were created.')
+    if attack_patterns:
+        demisto.debug(f'{len(attack_patterns)} Attack Patterns Indicators were created.')
+    if course_of_actions:
+        demisto.debug(f'{len(course_of_actions)} Course of Actions Indicators were created.')
+    if intrusion_sets:
+        demisto.debug(f'{len(intrusion_sets)} Intrusion Sets Indicators were created.')
```

Comment:
add module/display name to the log or clarity.

Reply:
done